src/main/resources/com/ncc/aif/restricted_hypothesis_aif.shacl

```diff
@@ -25,6 +25,59 @@ aida:SystemShape
         """ ;
     ] .
 
+#########################.;
+# 4.3 TA3 #4 Each hypothesis graph must have exactly one hypothesis importance value
+#------------------------
+aida:HypothesisImportanceRequiredShape
+    a sh:PropertyShape ;
+    sh:path aida:importance ;
+    sh:or (
+        [sh:propertyShape aida:HypothesisShape]
+        [sh:propertyShape aida:EventArgumentShape]
+        [sh:propertyShape aida:RelationArgumentShape]
+    ) ;
+    sh:minCount 1 ;
+    sh:maxCount 1 .
+
+aida:HypothesisShape
+    sh:property aida:HypothesisImportanceRequiredShape .
+
+#########################
+# 4.3 TA3 #5 Each event or relation (cluster) in the hypothesis must have exactly one event/relation importance value
+#------------------------
+aida:HypothesisEventRelationClusterImportanceRequiredShape
+    a sh:NodeShape ;
+
+    sh:target [
+        a sh:SPARQLTarget ;
+        sh:select """
+            PREFIX aida:  <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/InterchangeOntology#>
+            SELECT ?prototype
+            WHERE {
+                ?prototype a aida:SameAsCluster .
+                ?prototype aida:prototype/a ?type .
+                FILTER( ?type = aida:Event || ?type = aida:Relation )
+            }
+        """ ;
+    ];
+
+    # sh:maxCount 1 is inherited from aida_ontology.shacl
+    sh:property [
```

Comment:
Why not just use the property shape defined above? Also, this looks like aida:importance lives on the cluster. Is that what's expected. The issue says something about an "event/relation importance importance value".

Reply:
Same reasoning as above, I had issues with the test and was worried I was targeting more than just the importance values of other property nodes. I'll update this.